File: src/test/java/blackbox/writer/UnwritableWriter.java
package blackbox.writer

import java.io.IOException
import java.io.Writer

internal class UnwritableWriter : Writer() {
  @Throws(IOException::class)
  override fun write(cbuf: CharArray, off: Int, len: Int) {
    throw IOException("Cannot write")
  }

  @Throws(IOException::class)
  override fun flush() {
    throw IOException("Cannot flush")
  }

  @Throws(IOException::class)
  override fun close() {
    throw IOException("Cannot close")
  }
}

File: src/test/java/blackbox/writer/UnwritableWriter.kt
package blackbox.writer

import net.codinux.csv.kcsv.IOException
import java.io.Writer

internal class UnwritableWriter : Writer() {

  override fun write(cbuf: CharArray, off: Int, len: Int) {
    throw IOException("Cannot write")
  }

  override fun flush() {
    throw IOException("Cannot flush")
  }

  override fun close() {
    throw IOException("Cannot close")
  }
}